### ISSUE TITLE
Findings in redesigned forms

### ISSUE THREAD

**Issue by Wallabeng** [vmware/clarity#2681]
## Describe the bug
We are currently starting several projects and try to decide, whether to use the new forms or still use the old ones.
For this I compiled a form in stackblitz (https://stackblitz.com/edit/clarity-light-theme-v013-np2tgt) containing all inputs and tried to fit them in new forms with horizontal layout.
My goal is to use angular components only, without extra CSS.

Following are my findings for standard layout (no custom clr-col classes):
- Label in clr-radio-container isn't bold (missing ControlIdService on clr-radio-container)
- Toggle switch doesn't look correctly and needs CSS. #1878 doesn't mention it so I guess it's currently not part of this redesign and I just tried to use the old toggle in new forms layout?
- "Row labeling" a checkbox needs CSS, maybe this needs an additional wrapper/container like radio buttons? But as I see, according to #1878 this is still WiP
- Checkbox labels are rendered bold.
- Date is currently OK, as #2631 should fix it

When I try to use custom widths for label/input (second form in stackblitz), some more things seem to break as the generated control-containers won't take over the clr-col classes. I'm not sure if this is even supported with angular components or only possible with CSS components.

Please also correct me (or my stackblitz), when I use something wrong.

So after all would you recommend using the new forms from now on or wait until they're finished?

## How to reproduce

SB: https://stackblitz.com/edit/clarity-light-theme-v013-np2tgt

## Expected behavior

For every input mentioned in stackblitz, I want a label (in bold) and the input aligned and sized correctly.

## Versions

**App**

* Angular: 6.0.0
* Clarity: 0.13.2

**Device:**

* Type: MacBook
* OS: macOS High Sierra
* Browser Chrome
* Version 68.0.3440.106

**Comment by gnomeontherun** [comment]
Thanks for the details, since this is a large set of things it will take a little bit to review everything. However, here is a quick response.

1) This is a known issue, a fix is in the works but hasn't been opened as a PR yet.
2) Toggle switches haven't been updated yet, but there is a desire to update it someday.
3) Checkboxes haven't been implemented with new forms patterns yet, I'm working on this currently.
4) Datepicker working with new forms will be merged and released very soon.
5) Custom widths are being fixed in the next release (its been merged but not released) https://github.com/vmware/clarity/pull/2656

As far as the question 'should I use new forms or old', I would really hope that the new forms work well for you now (with the known bugs to be resolved). In my mind, the main issue is the lack of support on some form components (like checkboxes). However, the CSS styles for form controls work for now, so perhaps you can use them to get by until the Angular components are ready. They are a top priority to wrap up soon.

**Comment by Wallabeng** [comment]
Thanks for the quick update! I guess we will go with new forms using angular components where possible and CSS as fallback for now!

**Comment by Wallabeng** [comment]
I updated the stackblitz to current version 0.13.3 and I guess the custom widths are not working as (I?) expected.
If I put the label to md-4 and its input to md-8, I expected the control-container to render as md-8 and the input as md-12, to fill all of the control-containers' space.

Additionally it seems radio buttons are not affected by the change at all.
I also added a radio group example of how I expect them to render in the end (labelled "Expected radio")
